tests/integration/targets/hcloud_iso_info/tasks/test.yml

```diff
@@ -69,3 +69,16 @@
       - result.hcloud_iso_info | list | count > 2
       - result.hcloud_iso_info | selectattr('architecture', 'equalto', 'x86') | list | count == 0
       - result.hcloud_iso_info | selectattr('architecture', 'equalto', 'arm') | list | count  > 2
+
+- name: Gather hcloud_iso_info with deprecation
+  hetzner.hcloud.hcloud_iso_info:
+    id: "{{ hcloud_iso_deprecated_id }}"
+  register: result
+- name: Verify hcloud_iso_info with correct id
+  ansible.builtin.assert:
+    that:
+      - result.hcloud_iso_info | list | count == 1
+      - result.hcloud_iso_info[0].id == "{{ hcloud_iso_deprecated_id }}"
+      - result.hcloud_iso_info[0].name == "{{ hcloud_iso_deprecated_name }}"
+      - result.hcloud_iso_info[0].deprecation.announced == "{{ hcloud_iso_deprecated_announced }}"
+      - result.hcloud_iso_info[0].deprecation.unavailable_after == "{{ hcloud_iso_deprecated_unavailable_after }}"
```

Comment:
Deprecated ISOs are removed from the API after their `unavailable_after` date as far as I know. This test will then start to fail after `2023-11-05`.

I dont think we have a good option to reliably test this with the actual API :(

Reply:
Hmm, sad, a `include_deprecated` flag could we have been useful. I will remove this test.